src/ModuleLoader.ts

```diff
@@ -59,8 +64,19 @@ function getIdMatcher<T extends Array<any>>(
 		return (id, ...args) => (!id.startsWith('\0') && option(id, ...args)) || false;
 	}
 	if (option) {
-		const ids = new Set(Array.isArray(option) ? option : option ? [option] : []);
-		return (id => ids.has(id)) as (id: string, ...args: T) => boolean;
+		const ids = new Set<string>();
+		const matchers: RegExp[] = [];
+		for (const value of Array.isArray(option) ? option : [option]) {
+			if (value instanceof RegExp) {
+				matchers.push(value);
+			} else {
+				ids.add(value);
+			}
+		}
```

Comment:
I optimized this slightly so that a pre-created optimized Set is used for all string ids while only regular expressions are treated as an array. Also, having a separate case for just a single RegExp was not really necessary as this case is handled by the general logic as well. Also it appears that the parser will make sure we only ever pass a boolean, an array or a function here so we can avoid an unneeded test.